tests/core/fs/test_fs_online_rwfile_and_sync.py

```diff
@@ -13,6 +15,47 @@
 PLAYGROUND_SIZE = BLOCK_SIZE * 10
 
 
+@pytest.mark.trio
+@pytest.mark.slow
+async def test_leak_opened_file(user_fs_online_state_machine, alice: LocalDevice):
+    LOGGER = structlog.getLogger(test_leak_opened_file.__name__)
+    import resource
+
+    resource.setrlimit(resource.RLIMIT_NOFILE, [1024, 1024])
+
+    class FSOnlineRwFileAndAsync(user_fs_online_state_machine):
```

Comment:
`user_fs_online_state_machine` is an hypothesis state machine. We'd better not use it for regular test.

Here what we are doing in a loop is 1) creating a backend & a userfs (hence it underlying local storage) 2) creating a workspace.

This can be achieve very easily using the regular backend_factory/user_fs_factory fixtures.

However I wonder if this test is not too broad: checking the leak of file descriptor is only needed for component that actually interact with the outside world (remembre the `platform_...` crates in oxidation ? ^^).

So I think the test would be simpler by only testing file descriptor leakage on the storage component (for instance the lowest-level `LocalDatabase` class, considering all the other components will use it with a valid async context manager and hence we can trust they won't cause a leak unless the code is very very broken)

Reply:
You're right, currently the test is a snipped that is able to detected a fd leaks. When we known from where the leak come from, we could create a test specifically for it.